vaccine_feed_ingest/apis/placekey.py

```diff
@@ -29,53 +29,105 @@ def lookup_placekey(
-
-        if "error" in response:
-            logger.info("Failed to add placekey because: %s", response["error"])
-            self.set_with_expire(cache_key, {"error": response["error"]})
-            return None
+        return results.get("record_0")
 
-        placekey_id = response.get("placekey")
-
-        self.set_with_expire(cache_key, {"placekey": placekey_id})
-
-        return placekey_id
+    def lookup_placekeys(
+        self,
+        records: Dict[str, dict],
+        strict_address_match: bool = False,
+        strict_name_match: bool = False,
+    ) -> Dict[str, Optional[str]]:
+        result: Dict[str, Optional[str]] = {}
+
+        for record_id, record in records.items():
+            if not (latitude := record.get("latitude")):
+                continue
+            if not (longitude := record.get("longitude")):
+                continue
+            if not (location_name := record.get("location_name")):
+                continue
+            if not (street_address := record.get("street_address")):
+                continue
+            if not (city := record.get("city")):
+                continue
+            if not (region := record.get("region")):
+                continue
+            if not (postal_code := record.get("postal_code")):
+                continue
+            if not (iso_country_code := record.get("iso_country_code", "US")):
+                continue
+
+            cache_key = calculate_cache_key(
+                "placekey",
+                [
+                    f"{latitude:.5f}",
+                    f"{longitude:.5f}",
+                    location_name,
+                    street_address,
+                    city,
+                    region,
+                    postal_code,
+                    iso_country_code,
+                    str(strict_address_match),
+                    str(strict_name_match),
+                ],
+            )
+
+            cache_response = self.api_cache.get(cache_key)
+
+            if cache_response:
+                if "error" in cache_response:
+                    result[record_id] = None
+                    continue
+
+                result[record_id] = cache_response.get("placekey")
+                continue
+
+            response = self._placekey_api.lookup_placekey(
+                latitude=latitude,
+                longitude=longitude,
+                location_name=location_name,
+                street_address=street_address,
+                city=city,
+                region=region,
+                postal_code=postal_code,
+                iso_country_code=iso_country_code,
+                strict_address_match=strict_address_match,
+                strict_name_match=strict_name_match,
+            )
+
+            if not response:
+                result[record_id] = None
+                continue
+
+            if "error" in response:
+                logger.info("Failed to add placekey because: %s", response["error"])
```

Comment:
There are several other failure cases. Should we log before `continue`-ing (or `return`-ing) in them?

Reply:
Added